invenio_records_resources/resources/files/resource.py

```diff
@@ -174,6 +206,35 @@ def read_content(self):
 
         return item.send_file(), 200
 
+    @request_view_args
+    def read_archive(self):
+        """Read a zipped version of all files."""
+        all_in_one_enabled = current_app.config.get("ALL_IN_ONE_DOWNLOAD_ENABLED", True)
+        if not all_in_one_enabled:
+            abort(404)
```

Comment:
minor: instead of checking here, we could conditionally add the endpoint similarly as done with the `allow_upload` resource config variable (see my comment above in the routes).